WF-CDPInstaller/plugin_dir/plugin_installer.py

```diff
@@ -0,0 +1,186 @@
+import common.install_utils as utils
+import common.config as config
+import plugin_dir.plugin_exception as ex
+
+
+class PluginInstaller(object):
+    """
+    The interface for installers
+
+    The following methods need to be implemented
+      title():
+          - the ascii art for the installer
+      overview():
+          - general description about what the installer does
+      check_depedency()
+          - check plugin and installer's dependency
+      write_plugin():
+          - write the actual plugin file
```

Comment:
This will allow `output_config()` to do some validation and error correction on its input data.